$MirOS: ports/mailnews/pine/patches/patch-imap_src_osdep_unix_Makefile,v 1.5 2006/03/19 18:37:25 tg Exp $
$OpenBSD: patch-src_osdep_unix_Makefile,v 1.5 2003/10/05 22:00:00 jakob Exp $
$OpenBSD: patch-imap_src_osdep_unix_Makefile,v 1.2 2003/10/05 21:42:29 jakob Exp $

Second chunk must contain @ETC@ once

--- imap/src/osdep/unix/Makefile.orig	Sat Apr 30 20:50:51 2005
+++ imap/src/osdep/unix/Makefile	Thu Oct 18 16:16:13 2012
@@ -24,7 +24,7 @@ EXTRAAUTHENTICATORS=
 EXTRADRIVERS=mbox
 PASSWDTYPE=std
 SSLTYPE=nopwd
-IP=4
+IP=6
 
 
 # The optimization level here for GCC ports is set here for a reason.  It's
@@ -41,8 +41,8 @@ GCCOPTLEVEL= -O2
 
 # Extended flags needed for SSL.  You may need to modify.
 
-SSLDIR=/usr/local/ssl
-SSLCERTS=$(SSLDIR)/certs
+SSLDIR=/usr
+SSLCERTS=@ETC@/ssl
 SSLKEYS=$(SSLCERTS)
 SSLINCLUDE=$(SSLDIR)/include
 SSLLIB=$(SSLDIR)/lib
@@ -88,7 +88,7 @@ SPOOLDIR=/usr/spool
 MAILSPOOL=$(SPOOLDIR)/mail
 NEWSSPOOL=$(SPOOLDIR)/news
 RSHPATH=/usr/ucb/rsh
-LOCKPGM=/etc/mlock
+LOCKPGM=${PREFIX}/libexec/mlock
 
 
 # Default formats for creating new mailboxes and for empty mailboxes in the
@@ -107,11 +107,12 @@ LOCKPGM=/etc/mlock
 CREATEPROTO=unixproto
 EMPTYPROTO=unixproto
 
+.SUFFIXES: .o .so
 
 # Commands possibly overriden by the individual port
 
 ARRC=ar rc
-CC=cc
+CC?=cc
 LN=ln -s
 RANLIB=ranlib
 
@@ -131,6 +132,9 @@ BINARIES=osdep.o mail.o misc.o newsrc.o 
  unix.o mbx.o mmdf.o tenex.o mtx.o news.o phile.o mh.o mx.o
 CFLAGS=-g
 
+PICARCHIVE=c-client_pic.a
+PICBINARIES=${BINARIES:.o=.so}
+
 CAT=cat
 MAKE=make
 MV=mv
@@ -242,13 +246,13 @@ bsi:	# BSD/i386
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g $(GCCOPTLEVEL) -pipe"
 
-bso:	# OpenBSD
+bso:	# OpenBSD / MirOS BSD
 	$(BUILD) `$(CAT) SPECIALS` OS=bsi \
-	 SIGTYPE=psx CRXTYPE=nfs \
+	 SIGTYPE=psx CRXTYPE=nfs CHECKPW=bso \
 	 SPOOLDIR=/var \
-	 ACTIVEFILE=/usr/local/news/lib/active \
+	 ACTIVEFILE=${PREFIX}/news/lib/active \
 	 RSHPATH=/usr/bin/rsh \
-	 BASECFLAGS="$(GCCOPTLEVEL) -pipe"
+	 BASECFLAGS="$(GCCOPTLEVEL) -pipe -DBSO_AUTH"
 
 cvx:	# Convex
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
@@ -801,23 +805,38 @@ vu2:	# VAX Ultrix 2.3, etc.
 
 # Build it!
 
-build:	clean once $(ARCHIVE)
+NO_SHARED_LIBS?=no
+.if ${NO_SHARED_LIBS:L} == "yes"
+B_PICARCHIVE=
+.else
+B_PICARCHIVE=${PICARCHIVE}
+.endif
 
-all:	$(ARCHIVE)
+build:	clean once $(ARCHIVE) ${B_PICARCHIVE}
 
+all:	$(ARCHIVE) ${B_PICARCHIVE}
+
 $(ARCHIVE): $(BINARIES)
 	sh -c '$(RM) $(ARCHIVE) || true'
 	@$(CAT) ARCHIVE
 	@$(SH) ARCHIVE
 
+$(PICARCHIVE): $(PICBINARIES)
+	sh -c '$(RM) $(PICARCHIVE) || true'
+	@$(CAT) PICARCHIVE
+	@$(SH) PICARCHIVE
+
 .c.o:
 	`$(CAT) CCTYPE` -c `$(CAT) CFLAGS` $*.c
 
+.c.so:
+	`$(CAT) CCTYPE` -c `$(CAT) CFLAGS` ${PICFLAG} -DPIC -o $@ ${@:.so=.c}
 
+
 # Cleanup
 
 clean:
-	sh -c '$(RM) auths.c crexcl.c nfstest.c linkage.[ch] siglocal.c osdep*.[ch] *.o ARCHIVE *FLAGS *TYPE $(ARCHIVE) || true'
+	sh -c '$(RM) auths.c crexcl.c nfstest.c linkage.[ch] siglocal.c osdep*.[ch] *.o *.so ARCHIVE PICARCHIVE *FLAGS *TYPE $(ARCHIVE) ${PICARCHIVE} || true'
 
 
 # Dependencies
@@ -871,6 +890,9 @@ osdep.o:mail.h misc.h env.h fs.h ftl.h n
 	@echo or build with command: make `$(CAT) OSTYPE` SSLTYPE=none
 	`$(CAT) CCTYPE` -c `$(CAT) CFLAGS` `$(CAT) OSCFLAGS` -c osdep.c
 
+osdep.so: osdep.o
+	`$(CAT) CCTYPE` -c `$(CAT) CFLAGS` `$(CAT) OSCFLAGS` ${PICFLAG} -DPIC -c -o $@ osdep.c
+
 osdep.c: osdepbas.c osdepckp.c osdeplog.c osdepssl.c
 	$(CAT) osdepbas.c osdepckp.c osdeplog.c osdepssl.c > osdep.c
 
@@ -925,6 +947,7 @@ onceenv:
 	 -DRSHPATH=\"$(RSHPATH)\" -DLOCKPGM=\"$(LOCKPGM)\" > OSCFLAGS
 	echo $(BASELDFLAGS) $(EXTRALDFLAGS) > LDFLAGS
 	echo "$(ARRC) $(ARCHIVE) $(BINARIES);$(RANLIB) $(ARCHIVE)" > ARCHIVE
+	echo "$(ARRC) $(PICARCHIVE) $(PICBINARIES);$(RANLIB) $(PICARCHIVE)" > PICARCHIVE
 	echo $(OS) > OSTYPE
 	./drivers $(EXTRADRIVERS) $(DEFAULTDRIVERS) dummy
 	./mkauths $(EXTRAAUTHENTICATORS) $(DEFAULTAUTHENTICATORS)
